engine/src/main/java/org/terasology/engine/world/block/BlockUri.java

```diff
@@ -156,7 +162,8 @@ public boolean equals(Object obj) {
         }
         if (obj instanceof BlockUri) {
             BlockUri other = (BlockUri) obj;
-            return Objects.equal(other.blockFamilyDefinition, blockFamilyDefinition)
+            return this.hashCode == ((BlockUri) obj).hashCode
```

Comment:
The has code is computed from `blockFamilyDefinition`, `shape`, and `blockName`. If we are reasonably sure that we don't have hash collisions here, why check the other fields explicitly in addition?

Reply:
```suggestion
            return this.hashCode == other.hashCode
```

Reply:
the comparison falls through most of the time so the cost of checking the other terms is trivial.